### PR TITLE
Test:  Added e2e test for Work resource deletion.

### PR DESCRIPTION
**Pull request by Ealianis**
### Description of your changes

Added an e2e test for Work resource deletion that can occur on the Hub.

I have:
- [x] Read and followed Caravel's [Code of conduct](https://github.com/Azure/k8s-work-api/blob/master/code-of-conduct.md).
- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested
Local 'make test-e2e' results in 100% success.

### CODE REVIEW

File: tests/e2e/suite_test.go [hunk 1]
```diff
@@ -84,5 +85,8 @@ var _ = ginkgo.BeforeSuite(func() {
 	hubWorkClient, err = workclientset.NewForConfig(restConfig)
 	gomega.Expect(err).ToNot(gomega.HaveOccurred())
 
+	spokeWorkClient, err = workclientset.NewForConfig(restConfig)
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
```
Comment (aagusuab) [inline tests/e2e/suite_test.go:89]: Duplicate code

**Review by aagusuab — APPROVED** [review]
one comment. Other than that, looks good to me.

File: tests/e2e/suite_test.go (hunk 1 above)
Comment (Ealianis) [reply tests/e2e/suite_test.go:89]: Fixed.

File: tests/e2e/apply_test.go [hunk 2]
```diff
@@ -128,6 +131,54 @@ var _ = ginkgo.Describe("Apply Work", func() {
 			}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
 		})
 	})
+	ginkgo.Context("Deleting a Work resource from the Hub", func() {
+		ginkgo.It("should delete: Work resource, AppliedWork resource, and the actual resources created by the Work.spec", func() {
```
Comment (ryanzhang-oss) [inline tests/e2e/apply_test.go:135]: I know it's what's in the file now but why create a nested case for one test?

File: tests/e2e/apply_test.go [hunk 3]
```diff
@@ -128,6 +131,54 @@ var _ = ginkgo.Describe("Apply Work", func() {
...
+
+			// Get AppliedWork(s) so we can verify garbage collection.
+			aw, err := spokeWorkClient.MulticlusterV1alpha1().AppliedWorks().Get(context.Background(), workName, metav1.GetOptions{})
```
Comment (ryanzhang-oss) [inline tests/e2e/apply_test.go:138]: when is the work applied?

Comment (Ealianis) [reply tests/e2e/apply_test.go:138]: Within the wrapping Describe block. I'm using the nested hierarchy to encapsulate and repurpose a series of tests that are somewhat sequential in nature. There is a current data state dependency within the contexts nested within a describe, that state being Work (A)

e.g.
- Test 1 -> Creates Work (A)
- Test 2 -> Modifies Work (A)
- Test 3 -> Deletes Work (A)

 I believe this should be refactored to use BeforeEach / AfterEach at the context level to remove the test created data state dependencies.

File: tests/e2e/apply_test.go (hunk 2 above)
Comment (ryanzhang-oss) [reply tests/e2e/apply_test.go:135]: oh, the convention is to add a nested block only if that's a logic unit. There is no reason to have a context here.

File: tests/e2e/apply_test.go (hunk 3 above)
Comment (ryanzhang-oss) [reply tests/e2e/apply_test.go:138]: have dependence between tests is a common but also the most damaging anti-pattern in writing tests, please refactor that.

Comment (Ealianis) [reply tests/e2e/apply_test.go]: Done

**Pull request merged**